src/com/goide/inspections/GoNoNewVariablesInspection.java

```diff
@@ -41,16 +41,19 @@ protected GoVisitor buildGoVisitor(@NotNull final ProblemsHolder holder, @NotNul
     return new GoVisitor() {
       @Override
       public void visitShortVarDeclaration(@NotNull GoShortVarDeclaration o) {
+        super.visitShortVarDeclaration(o);
         visitVarDefinitionList(o, o.getVarDefinitionList());
       }
 
       @Override
       public void visitRecvStatement(@NotNull GoRecvStatement o) {
+        super.visitRecvStatement(o);
         visitVarDefinitionList(o, o.getVarDefinitionList());
       }
 
       @Override
       public void visitRangeClause(@NotNull GoRangeClause o) {
+        super.visitRangeClause(o);
```

Comment:
it's easier to invoke checkCanceld in visitVarDefinitionList and remove all supers